src/main/java/com/atlassian/jira/cloud/jenkins/util/CompoundIssueKeyExtractor.java

```diff
@@ -0,0 +1,27 @@
+package com.atlassian.jira.cloud.jenkins.util;
+
+import com.atlassian.jira.cloud.jenkins.common.service.IssueKeyExtractor;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Combines multiple issue key extractors. */
+public class CompoundIssueKeyExtractor implements IssueKeyExtractor {
```

Comment:
Instead of passing around lists of issue extractors, I built an extractor that combines multiple extractors so we can pass around a single one